chapters/en/chapter5/fine-tuning.mdx

```diff
@@ -0,0 +1,549 @@
+
+```python
+from functools import partial
+
+# disable cache during training since it's incompatible with gradient checkpointing
+model.config.use_cache = False
+
+# set language and task for generation
+model.generation_config.language = "sinhalese"
+model.generation_config.task = "transcribe"
+
+# re-enable cache for inference
+model.generate = partial(model.generate, use_cache=True)
+```
+
+## Define the Training Configuration
+
+In the final step, we define all the parameters related to training. Here, you can set the `max_steps` to train for longer.
+For more detail on the training arguments, refer to the Seq2SeqTrainingArguments [docs](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.Seq2SeqTrainingArguments).
+
+```python
+from transformers import Seq2SeqTrainingArguments
+
+training_args = Seq2SeqTrainingArguments(
+    output_dir="./whisper-small-dv",  # name on the HF Hub
+    per_device_train_batch_size=16,
+    gradient_accumulation_steps=1,  # increase by 2x for every 2x decrease in batch size
+    learning_rate=1e-5,
+    warmup_steps=500,
+    max_steps=4000,
+    gradient_checkpointing=True,
+    fp16=True,
+    evaluation_strategy="steps",
+    per_device_eval_batch_size=8,
+    predict_with_generate=True,
+    generation_max_length=225,
+    save_steps=1000,
+    eval_steps=1000,
+    logging_steps=25,
+    report_to=["tensorboard"],
+    load_best_model_at_end=True,
+    metric_for_best_model="wer",
+    greater_is_better=False,
+    push_to_hub=True,
+)
+```
+
+<Tip>
+    If you do not want to upload the model checkpoints to the Hub, set `push_to_hub=False`.
+</Tip>
+
+We can forward the training arguments to the 🤗 Trainer along with our model, dataset, data collator and `compute_metrics` function:
+
+```python
+from transformers import Seq2SeqTrainer
+
+trainer = Seq2SeqTrainer(
+    args=training_args,
+    model=model,
+    train_dataset=common_voice["train"],
+    eval_dataset=common_voice["test"],
+    data_collator=data_collator,
+    compute_metrics=compute_metrics,
+    tokenizer=processor,
+)
+```
+
+And with that, we're ready to start training!
+
+### Training
+
+To launch training, simply execute:
+
+```python
+trainer.train()
+```
+
+Training will take approximately 5-10 hours depending on your GPU or the one allocated to the Google Colab. Depending on
+your GPU, it is possible that you will encounter a CUDA `"out-of-memory"` error when you start training. In this case,
+you can reduce the `per_device_train_batch_size` incrementally by factors of 2 and employ [`gradient_accumulation_steps`](https://huggingface.co/docs/transformers/main_classes/trainer#transformers.Seq2SeqTrainingArguments.gradient_accumulation_steps)
+to compensate.
+
+**Output:**
+
+| Step | Training Loss | Epoch | Validation Loss |  WER  |
+|:----:|:-------------:|:-----:|:---------------:|:-----:|
+
+Our best WER is X% TODO(SG) - not bad for 7h of training data (I hope...)! The big question is how this
```

Comment:
Missing WER

Reply:
Resolved in https://github.com/huggingface/audio-transformers-course/pull/70/commits/9171b6c80f0b4f0fbd382eb75cf2336a919683b6